tools/prune_db.py

```diff
@@ -19,15 +20,16 @@
         print("This will remove ALL runs, measurement, CI, carbonDB and hog data from the DB. Continue? (y/N)")
         answer = sys.stdin.readline()
         if answer.strip().lower() == 'y':
-            DB().query('TRUNCATE runs CASCADE')
-            DB().query('TRUNCATE ci_measurements CASCADE')
-            DB().query('TRUNCATE hog_measurements CASCADE')
-            DB().query('TRUNCATE carbondb_energy_data CASCADE')
-            DB().query('TRUNCATE carbondb_energy_data_day CASCADE')
+            tables = ['runs', 'ci_measurements', 'hog_measurements', 'carbondb_data', 'carbondb_data_raw']
+            for table in tables:
+                try:
+                    DB().query(f"TRUNCATE TABLE {table} CASCADE")
+                except errors.UndefinedTable:
+                    continue
             print("Done")
     elif args.mode == 'failed-runs':
         print("This will remove all runs that have not ended, which includes failed ones, but also possibly running, so be sure no measurement is currently active. Continue? (y/N)")
         answer = sys.stdin.readline()
         if answer.strip().lower() == 'y':
-            DB().query('DELETE FROM runs WHERE end_measurement IS NULL')
+            DB().query('DELETE FROM runs WHERE failed = TRUE')
```

Comment:
Did you experience issues with using `end_measurement is NULL`. At the moment the reality in the database is often that runs that had hard hickups do not get failed = TRUE set as the DB failed before that. But not having end_measurement is more reliable.

Reply:
Yes, I had measurement runs in the database that ran until the end, but were correctly marked as failed. The attribute `end_measurement` was therefore not NULL and the script was not able to delete these failed measurements from the database.

So the query should be changed to `DELETE FROM runs WHERE failed = TRUE OR end_measurement IS NULL`?

Reply:
jap, that would be a better condition then.

This DB field is due for a refactor in the near future anyway. But checking for `failed = TRUE OR end_measurement IS NULL ` should be the best for now.